htlcswitch/link.go

```diff
@@ -1725,15 +1725,18 @@ func (l *channelLink) processLockedInHtlcs(
 					fwdInfo.AmountToForward,
 				)
 
-				// If the amount of the incoming HTLC, minus
-				// our expected fee isn't equal to the
-				// forwarding instructions, then either the
-				// values have been tampered with, or the send
-				// used incorrect/dated information to
+				// If the actual fee is less than our expected
+				// fee, then we'll reject this HTLC as it
+				// didn't provide a sufficient amount of fees,
+				// or the values have been tampered with, or
+				// the send used incorrect/dated information to
 				// construct the forwarding information for
 				// this hop. In any case, we'll cancel this
 				// HTLC.
-				if pd.Amount-expectedFee < fwdInfo.AmountToForward {
+				actualFee := pd.Amount - fwdInfo.AmountToForward
+				if actualFee < expectedFee ||
```

Comment:
If `pd.Amount == fwdInfo.AmountToForward`, the I believe this will fail?

Reply:
Atm, as as we assume we _always_ take a fee. I can modify to relax the case perhaps?

Reply:
Fixed in fixup commit just pushed.